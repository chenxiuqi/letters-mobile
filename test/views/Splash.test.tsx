import * as React from "react";
import { SplashScreen } from "@views";
import renderer from "react-test-renderer";
import { Image } from "react-native";

const AmeelioLogo = require("@assets/Ameelio_Logo.png");

describe("Splash screen", () => {
  it("should render", () => {
    const tree = renderer.create(<SplashScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should show the logo", () => {
    const splash = renderer.create(<SplashScreen />).root;
    const img = splash.findByType(Image);
    expect(img.props.source).toEqual(AmeelioLogo);
  });
});
