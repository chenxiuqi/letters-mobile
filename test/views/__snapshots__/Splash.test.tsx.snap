// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      accessibilityLabel="AmeelioLogo"
      source={1}
      style={
        Object {
          "aspectRatio": 2.2768361581920904,
          "height": 120,
        }
      }
    />
  </View>
</View>
`;
